exporters/otlp/otlptrace/internal/otlptracetest/clienttest.go

```diff
@@ -0,0 +1,120 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package otlptracetest
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"testing"
+	"time"
+
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
+)
+
+func RunExporterShutdownTest(t *testing.T, factory func() otlptrace.Client) {
+	t.Run("testClientStopHonorsTimeout", func(t *testing.T) {
+		testClientStopHonorsTimeout(t, factory())
+	})
+
+	t.Run("testClientStopHonorsCancel", func(t *testing.T) {
+		testClientStopHonorsCancel(t, factory())
+	})
+
+	t.Run("testClientStopNoError", func(t *testing.T) {
+		testClientStopNoError(t, factory())
+	})
+
+	t.Run("testClientStopManyTimes", func(t *testing.T) {
+		testClientStopManyTimes(t, factory())
+	})
+}
+
+func initializeExporter(t *testing.T, client otlptrace.Client) *otlptrace.Exporter {
+	ctx := context.Background()
+	e, err := otlptrace.NewExporter(ctx, client)
+	if err != nil {
+		t.Fatalf("failed to create exporter")
+	}
+
+	return e
+}
+
+func testClientStopHonorsTimeout(t *testing.T, client otlptrace.Client) {
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
```

Comment:
What's the point of these "outter" context? They seem superfluous.

Reply:
It was supposed to be used on the call to the `NewExporter()`, since I've created a `initializeExporter` to do create an exporter, I've moved this context to that function as well.